### PR TITLE
Core 2.0 style update

### PR DESCRIPTION
**Pull request by xperiandri**
Fixed issues with missing usings in cshtml files
Updated Web project configuration to Core 2.0 style

### CODE REVIEW

**Comment by xperiandri** [comment]
@sebastienros, looks like tests need some tweaks too, don't they?

File: src/Orchard.Cms.Web/Program.cs [hunk 1]
```diff
@@ -1,20 +1,50 @@
...
+                    }
+
+                    config.AddEnvironmentVariables();
+
+                    if (args?.Any() ?? false) config.AddCommandLine(args);
+                })
+                .ConfigureLogging((hostingContext, logging) =>
+                {
+                    logging.AddConfiguration(hostingContext.Configuration.GetSection("Logging"));
+                    logging.AddConsole();
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Program.cs]: Nope, killing perf. Should be in dev only. If it was not there don't add it.

File: src/Orchard.Cms.Web/Program.cs [hunk 2]
```diff
@@ -1,20 +1,50 @@
...
+                    logging.AddDebug();
+                    logging.AddNLogWeb(hostingContext.HostingEnvironment);
+                })
                 .UseStartup<Startup>()
+                .UseApplicationInsights()
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Program.cs]: Why did you add it?

File: src/Orchard.Cms.Web/Program.cs [hunk 3]
```diff
@@ -1,20 +1,50 @@
...
+                {
+                    var env = hostingContext.HostingEnvironment;
+                    config.AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                          .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true)
+                          .AddJsonFile("logging.json", optional: true, reloadOnChange: true)
+                          .AddJsonFile($"logging.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
+
+                    if (env.IsDevelopment())
+                    {
+                        var appAssembly = Assembly.Load(new AssemblyName(env.ApplicationName));
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Program.cs]: What is that, and why just in dev?

File: src/Orchard.Cms.Web/Program.cs (hunk 2 above)
Comment (Jetski5822) [inline src/Orchard.Cms.Web/Program.cs]: Do we want this?

Comment (sebastienros) [inline src/Orchard.Cms.Web/Program.cs]: Would make sense as a module. Not here, not now (not a core 2.0 concern).

Comment (Jetski5822) [inline src/Orchard.Cms.Web/Program.cs]: I agree, need to do it on a different pr

Comment (xperiandri) [inline src/Orchard.Cms.Web/Program.cs]: Because Application Insights in Visual Studio write tracing into output and hide all the useful information.
The only way to disable it I know is to add their package set particular flag off

File: src/Orchard.Cms.Web/Program.cs (hunk 3 above)
Comment (xperiandri) [inline src/Orchard.Cms.Web/Program.cs]: This piece I copied from `CreateDefaultBuilder` implementation.
By the way, maybe it is better to copy all and do not call `CreateDefaultBuilder`, I don't know

File: src/Orchard.Cms.Web/Program.cs (hunk 1 above)
Comment (sebastienros) [inline src/Orchard.Cms.Web/Program.cs]: If that's the case then good, but for Console that would surprise me as it's very limited in terms of config, for instance no way to enable only specific loggers. Whatever solution is used, it can't be there by default.

File: src/Orchard.Cms.Web/Program.cs (hunk 2 above)
Comment (sebastienros) [inline src/Orchard.Cms.Web/Program.cs]: OMG

Comment (sebastienros) [inline src/Orchard.Cms.Web/Program.cs]: But doing that `UseApplicationInsight` will probably register ASP.NET MVC tag helper components including it in all pages!

Comment (xperiandri) [inline src/Orchard.Cms.Web/Program.cs]: No, not at all as far as I know. You have to inject that into layout root manually. I did that manually last time I enabled it.

File: src/Orchard.Cms.Web/Program.cs [hunk 4]
```diff
@@ -1,20 +1,50 @@
...
     public class Program
     {
         public static void Main(string[] args)
         {
-            var host = new WebHostBuilder()
-                .UseIISIntegration()
-                .UseKestrel()
-                .UseContentRoot(Directory.GetCurrentDirectory())
+            var host = WebHost.CreateDefaultBuilder(args)
+                .ConfigureAppConfiguration((hostingContext, config) =>
```
Comment (PinpointTownes) [inline src/Orchard.Cms.Web/Program.cs]: Out of curiosity, why do you need to do all these things here? Adding the `appsettings` files or enabling the user secrets are things `CreateDefaultBuilder()` already does for you.

File: src/Orchard.Cms.Web/Program.cs (hunk 1 above)
Comment (PinpointTownes) [inline src/Orchard.Cms.Web/Program.cs]: I wouldn't be surprised if we ended up with 2 console loggers with this change (it's already registered by `CreateDefaultBuilder()`) :sweat_smile:

File: src/Orchard.Cms.Web/Program.cs (hunk 2 above)
Comment (PinpointTownes) [inline src/Orchard.Cms.Web/Program.cs]: Not sure registering it will help during the debug stage, it's automatically registered by VS at runtime, anyway :sweat_smile:

File: src/OrchardCore/Orchard.Logging.NLog/LoggingBuilderExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+#pragma warning disable RCS1075 // Avoid empty catch clause that catches System.Exception.
+            catch (Exception)
+#pragma warning restore RCS1075 // Avoid empty catch clause that catches System.Exception.
+            {
+                //ignore
+            }
+
+            LogManager.AddHiddenAssembly(typeof(ConfigureExtensions).GetTypeInfo().Assembly);
+
+            using (var provider = new NLogLoggerProvider(options))
```
Comment (PinpointTownes) [inline src/OrchardCore/Orchard.Logging.NLog/LoggingBuilderExtensions.cs]: I have a strange feeling. Will adding a logger provider that will be disposed just after `AddProvider` returns really work as you expect? :sweat_smile:

Comment (xperiandri) [inline src/OrchardCore/Orchard.Logging.NLog/LoggingBuilderExtensions.cs]: I copied this from source. Just modified from IloggingFactory to ILoggingBuilder.
Author does not have any remarks
https://github.com/NLog/NLog.Extensions.Logging/issues/108

File: src/Orchard.Cms.Web/Program.cs (hunk 1 above)
Comment (xperiandri) [inline src/Orchard.Cms.Web/Program.cs]: Don't they override?
@PinpointTownes, do you think that if `CreateDefaultBuilder()` is not satisfactory then it must not be called but its modified source used?

File: src/Orchard.Cms.Web/Program.cs (hunk 2 above)
Comment (xperiandri) [inline src/Orchard.Cms.Web/Program.cs]: Maybe, I just wanted to set static property in the Application Insights library.
I'm not sure what happens if I set it but not call this.

File: src/Orchard.Cms.Web/Program.cs (hunk 1 above)
Comment (PinpointTownes) [inline src/Orchard.Cms.Web/Program.cs]: > Don't they override?

No, the delegates registered by the `ConfigureXYZ` methods are cumulative.

File: src/OrchardCore/Orchard.Logging.NLog/LoggingBuilderExtensions.cs (hunk 5 above)
Comment (PinpointTownes) [inline src/OrchardCore/Orchard.Logging.NLog/LoggingBuilderExtensions.cs]: > Author does not have any remarks

That works because `NLogLoggerProvider` implements `Dispose()` as a no-op method. If that changes in the future, we'll have an issue to fix in Orchard. Better to avoid it by removing the `using`.

Comment (xperiandri) [inline src/OrchardCore/Orchard.Logging.NLog/LoggingBuilderExtensions.cs]: @PinpointTownes do we need to repost your message here https://github.com/NLog/NLog.Extensions.Logging/issues/108 then?

**Comment by sebastienros** [comment]
@xperiandri would you mind splitting this PR in different ones? Logging; views; app insights. Some changes block others.

**Comment by sebastienros** [comment]
Also please target the dev branch

**Comment by sebastienros** [comment]
I have just fixed the view in dev

**Comment by xperiandri** [comment]
OK, I'll try.
So is dev now on Core 2.0?

**Comment by sebastienros** [comment]
yes
